### PR TITLE
Minor tweak to actually use closest origin point for timezone

### PR DESCRIPTION
**Pull request by PureW**

### CODE REVIEW

File: src/thor/astar.cc [hunk 1]
```diff
@@ -291,7 +291,7 @@ AStarPathAlgorithm::GetBestPath(valhalla::Location& origin,
     // invalid label indicates there are no edges that can be expanded.
     uint32_t predindex = adjacencylist_->pop();
     if (predindex == kInvalidLabel) {
-      LOG_ERROR("Route failed after iterations = " + std::to_string(edgelabels_.size()));
+      LOG_ERROR("Route failed after iterations = " + std::to_string(edgelabels_.size()) + " because of invalid edge-index");
```
Comment (kevinkreiser) [inline src/thor/astar.cc]: Invalid edge index is not really relevant to the error message. The comment above says exactly what it means. Basically invalid edge index just means its the adjacency list is empty, ie the search is exhausted and there arent any more edges to look at. I would favor a message more along those lines.

File: src/thor/astar.cc [hunk 2]
```diff
@@ -434,8 +437,11 @@ void AStarPathAlgorithm::SetOrigin(GraphReader& graphreader,
     }
 
     // Store the closest node info
-    if (closest_ni == nullptr) {
+    auto nodeinfo_point= nodeinfo->latlng(tile->BoundingBox().minpt());
+    float node_distance = origin_ll.DistanceSquared(nodeinfo_point) < distance_to_closest_ni;
```
Comment (kevinkreiser) [inline src/thor/astar.cc]: in the end this is only measuring the distance to the closest end node, if we wanted to be really pedantic about most appropriate timezone we should probably also consider the begin nodes of the edges. of course for extreme accuracy we'd have to load the timezone sqlite db and do point in polygon, but yeah lets not do that :D

File: src/thor/astar.cc [hunk 3]
```diff
@@ -434,8 +437,11 @@ void AStarPathAlgorithm::SetOrigin(GraphReader& graphreader,
     }
 
     // Store the closest node info
-    if (closest_ni == nullptr) {
+    auto nodeinfo_point= nodeinfo->latlng(tile->BoundingBox().minpt());
```
Comment (kevinkreiser) [inline src/thor/astar.cc]: how does the linter miss this?

**Comment by kevinkreiser** [comment]
@PureW looks like build-release-x86 is missing the linter check for fast build failure. we should add that

**Comment by PureW** [comment]
> @PureW looks like build-release-x86 is missing the linter check for fast build failure. we should add that

I went down the rabbit hole, but in the end, `mason` does not seem to provide 32bit executables.

File: src/thor/astar.cc (hunk 2 above)
Comment (PureW) [reply src/thor/astar.cc]: I'm happy to scrap this more complicated logic I added and instead update the comment `// Store the closest node info` to say something along _grab the first nodeinfo for timezone estimation, exactly closest node is not important_.

How's that?

Comment (kevinkreiser) [reply src/thor/astar.cc]: sounds good to me. i just wanted to point the complexity out, in case you werent aware

**Pull request merged**